core/src/main/java/jenkins/util/io/PathRemover.java

```diff
@@ -61,7 +61,7 @@ public static PathRemover newRemoverWithStrategy(@Nonnull RetryStrategy retryStr
     }
 
     public static PathRemover newFilteredRobustRemover(@Nonnull PathChecker pathChecker, int maxRetries, boolean gcAfterFailedRemove, long waitBetweenRetries) {
-        return new PathRemover(new PausingGCRetryStrategy(Math.max(maxRetries, 1), gcAfterFailedRemove, waitBetweenRetries), pathChecker);
+        return new PathRemover(new PausingGCRetryStrategy(Math.max(maxRetries, 0), gcAfterFailedRemove, waitBetweenRetries), pathChecker);
```

Comment:
should be able to use the RobustRemover without any retries (ie not robust) by just changing some config without having to perform an if / else and calling 2 different code paths.